### PR TITLE
How to connect to Timescale Cloud with stricter SSL

### LINKED ISSUE #1597: [Docs RFC] Add signed certificate information, ssl connection options, and root cert bundle steps
Currently Timescale services in Cloud contain a self-signed certificate within the certificate chain. This situation means that you cannot connect certain services to a Timescale Cloud db service like a BI tool (PowerBI, Apache Superset) which require a fully signed certificate chain. Essentially if your web browser could connect to a Cloud postgres service right now it would complain like it does when it "doesn't trust the connection".  

We have a new system that fixes this issue. Every new Cloud service initially launches very quickly with a self-signed certificate. In the background (asynchronously) a signed certificate is requested for the service and then the service is updated with very little interruption. Connections to the service will be reset when the service is updated to a fully signed certificate chain but a simple reconnect will fix that and most connection services do this automatically. 

Beyond enabling more services to seamlessly connect to Cloud db services this also allows customers to user more secure connection configurations via the `sslmode` connection string flag.

Step 1.

```shell
openssl s_client -showcerts -partial_chain -starttls postgres \
                 -connect $SERVICE_URL_WITH_PORT < /dev/null 2>/dev/null | \
                 awk '/BEGIN CERTIFICATE/,/END CERTIFICATE/{ print }' > bundle.crt
# macos
pbcopy < bundle.crt
```

Step 2.
Paste the bundle (`bundle.crt`) which should be in your clipboard now into the large text area for https://whatsmychaincert.com/ to generate a correct chain. Make sure check the box below the text area for “Include Root Certificate”

whatsmychaincert create bundle

Step 3.
Save the bundle which downloaded from https://whatsmychaincert.com/ into `~/.postgresql/root.crt`

Step 4.
Connect to your Timescale Cloud instance using a more strict SSL connection modes (`sslmode`) such as: `verify-ca` or the highest level `verify-full` (see all options in the [SSL Support docs](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-SSLMODE-STATEMENTS).

For example:
`psql "postgres://tsdbadmin@$SERVICE_URL_WITH_PORT/tsdb?sslmode=verify-full"`

Or:
`psql "postgres://tsdbadmin@$SERVICE_URL_WITH_PORT/tsdb?sslmode=verify-ca"`

☝🏼 both of with require the `~/.postgresql/root.crt` file correctly generated as described

Instead of:
`psql "postgres://tsdbadmin@$SERVICE_URL_WITH_PORT/tsdb?sslmode=require"`

Which does not require these steps listed above.

# Subject matter expert (SME)

@minkimipt 

# Deadline

ASAP 😬 

# Any further info

Timescale Internal Tracking link https://github.com/timescale/product/issues/132

### PR DESCRIPTION
**Pull request by charislam**
You can now connect to Timescale Cloud with a stricter SSL mode that verifies the certificate chain.

# Links

Fixes #1597 

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

**Comment by minkimipt** [comment]
@charislam is it already expected to be reachable. I can't reach it.
image

**Comment by charislam** [comment]
@minkimipt 😞 It should! (worked on my browser, have you tried incognito mode?)

If it still doesn't work for you here's a screenshot you can look at in the meantime:
image

**Review by minkimipt — APPROVED** [review]
Approving it because the procedure is correct. Although I think we need more explanation about the difference of self signed certificates and proper ones and why it's all needed. I provided some suggestions.

File: cloud/page-index/page-index.js [hunk 1]
```diff
@@ -220,9 +220,25 @@ module.exports = [
...
+            href: "overview",
+            tags: ["encryption", "VPC", "privacy"],
+            keywords: ["security"],
+            excerpt: "Get an overview of Timescale Cloud security",
+          },
+          {
+            title: "Connect with a stricter SLL mode",
+            href: "strict-ssl",
+            tags: ["security", "certificates"],
+            keywords: ["ssl"],
```
Comment (clarkbw) [inline cloud/page-index/page-index.js:237]:
```suggestion
            keywords: ["ssl", "sslmode"],
```
`sslmode` is probably a keyword here as well.

File: cloud/security/index.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+---
+
+# Timescale Cloud security
+
+Learn how your Timescale Cloud instance is secured to protect your data and
+privacy.
+
+*   Get an [overview of Timescale Cloud security][overview]
+*   Learn how to [connect with a stricter SSL mode][ssl] for enhanced protection
+
```
Comment (clarkbw) [inline cloud/security/index.md:15]: Feels like this is the area we should add content like MST has in https://docs.timescale.com/mst/latest/security/

File: cloud/security/strict-ssl.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+keywords: [security]
+tags: [ssl]
+---
+
+# Connect with a stricter SSL mode
+
+The default connection string for Timescale Cloud uses the SSL mode `require`.
+If you want more security, you can connect with a SSL mode of `verify-ca` or
+`verify-full`. To do so, you need to store a copy of the certificate chain where
+your connection tool can find it.
```
Comment (clarkbw) [inline cloud/security/strict-ssl.md:14]: I wonder if we should be linking to the list of `sslmode` options here https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION

File: cloud/security/strict-ssl.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+This section provides instructions on setting up a stricter SSL connection.
+
+## SSL certificates on Timescale Cloud
+
+To spin up your service quickly, Timescale Cloud databases are started up with a
```
Comment (clarkbw) [inline cloud/security/strict-ssl.md:20]:
```suggestion
To spin up your service quickly, Timescale Cloud databases are initialized with a
```

File: cloud/security/strict-ssl.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+# Connect with a stricter SSL mode
+
+The default connection string for Timescale Cloud uses the SSL mode `require`.
+If you want more security, you can connect with a SSL mode of `verify-ca` or
+`verify-full`. To do so, you need to store a copy of the certificate chain where
+your connection tool can find it.
+
+This section provides instructions on setting up a stricter SSL connection.
+
+## SSL certificates on Timescale Cloud
```
[the remark this answers is not in the document]
Comment (clarkbw) [reply cloud/security/strict-ssl.md:18]: This is a lot of text 😮 I'd like to see this down to a succinct single paragraph or two. I'll take a pass at it today.

File: cloud/security/strict-ssl.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                 awk '/BEGIN CERTIFICATE/,/END CERTIFICATE/{ print }' > bundle.crt
+    ```
+
+1.  Copy the bundle to your clipboard. On MacOS, you can run:
+
+    ```shell
+    pbcopy < bundle.crt
+    ```
+
+1.  Navigate to <https://whatsmychaincert.com/>.
```
[the remark this answers is not in the document]
Comment (clarkbw) [reply cloud/security/strict-ssl.md:61]:
```suggestion
1.  Navigate to <https://whatsmychaincert.com/>. This online tool generates a full certificate chain including the root CA certificate which is not included in the bundle.
```

**Review by clarkbw — COMMENTED** [review]
I'll try to rewrite that longer section today. But feel free to take a pass at it before I can.

File: cloud/security/strict-ssl.md [hunk 7]
```diff
@@ -58,7 +64,9 @@ To set up a stricter SSL connection:
     pbcopy < bundle.crt
     ```
 
-1.  Navigate to <https://whatsmychaincert.com/>.
+1.  Navigate to <https://whatsmychaincert.com/>. This online tool generates a
+    full certificate chain, including the root CA certificate, which is not
+    included in the bundle.
```
Comment (minkimipt) [inline cloud/security/strict-ssl.md:69]:
```suggestion
    included in the certificate bundle returned by the database.
```

File: cloud/page-index/page-index.js [hunk 8]
```diff
@@ -220,9 +220,25 @@ module.exports = [
...
+            title: "Security overview",
+            href: "overview",
+            tags: ["encryption", "VPC", "privacy"],
+            keywords: ["security"],
+            excerpt: "Get an overview of Timescale Cloud security",
+          },
+          {
+            title: "Connect with a stricter SLL mode",
+            href: "strict-ssl",
+            tags: ["security", "certificates"],
```
Comment (charislam) [inline cloud/page-index/page-index.js:236]:
```suggestion
            tags: ["security", "certificates", "sslmode"],
```

File: cloud/security/strict-ssl.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+`verify-full`. To do so, you need to store a copy of the certificate chain where
+your connection tool can find it.
+
+This section provides instructions on setting up a stricter SSL connection.
+
+## SSL certificates on Timescale Cloud
+
+To spin up your service quickly, Timescale Cloud databases are started up with a
+self-signed certificate. After your service is started, a signed certificate is
+requested behind the scenes. When it's received, the certificate is replaced
```
[the remark this answers is not in the document]
Comment (minkimipt) [reply cloud/security/strict-ssl.md]: The only way users can make sure that proper certificates are being used is try to connect to their instance and inspect the certificate returned by the database. We can provide openssl command describing how that can be done.

Regarding the time that it takes for certificate to be issued and made available to users - it is usually not longer than 15 minutes could be faster but could also be slower if ZeroSSL backend is slow returning the certificate. That 15 minute interval is coupled with configuration of timescaledb operator and that could also be changed without notice. I think it's safe to write that certificate is going to be available within 30 minutes.

File: cloud/security/strict-ssl.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    ```shell
+    openssl s_client -showcerts -partial_chain -starttls postgres \
+                 -connect $SERVICE_URL_WITH_PORT < /dev/null 2>/dev/null | \
+                 awk '/BEGIN CERTIFICATE/,/END CERTIFICATE/{ print }' > bundle.crt
+    ```
+
+1.  Copy the bundle to your clipboard. On MacOS, you can run:
+
+    ```shell
+    pbcopy < bundle.crt
```
[the remark this answers is not in the document]
Comment (clarkbw) [reply cloud/security/strict-ssl.md:64]: via https://stackoverflow.com/questions/5130968/how-can-i-copy-the-output-of-a-command-directly-into-my-clipboard

```shell
# windows
clip.exe < bundle.crt

# linux
xclip -sel clip < bundle.crt
```

File: cloud/security/strict-ssl.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (clarkbw) [reply cloud/security/strict-ssl.md:19]: Looks great 👍🏼

File: cloud/security/index.md (hunk 2 above)
Comment (clarkbw) [reply cloud/security/index.md:15]: Ah ok! I didn't find it.

**Review by clarkbw — APPROVED** [review]

File: cloud/security/strict-ssl.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (charislam) [reply cloud/security/strict-ssl.md:64]: 🙇🏼

**Review by Loquacity — APPROVED** [review]

**Pull request merged**